tests/phpunit/Integration/JSONScript/TestCases/tree-01.json

```diff
@@ -0,0 +1,114 @@
+			"contents": "<!-- No parent -->[[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 3",
+			"contents": "<!-- No parent -->[[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 11",
+			"contents": "[[Has parent::Example/Tree 1]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 12",
+			"contents": "[[Has parent::Example/Tree 1]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 13",
+			"contents": "[[Has parent::Example/Tree 1]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 21",
+			"contents": "[[Has parent::Example/Tree 2]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 22",
+			"contents": "[[Has parent::Example/Tree 2]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 31",
+			"contents": "[[Has parent::Example/Tree 3]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 32",
+			"contents": "[[Has parent::Example/Tree 3]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 33",
+			"contents": "[[Has parent::Example/Tree 3]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 121",
+			"contents": "[[Has parent::Example/Tree 12]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 321",
+			"contents": "[[Has parent::Example/Tree 32]][[Part of::Tree test]]"
+		},
+		{
+			"page": "Example/Tree 01-01",
+			"contents": "{{#ask:[[Part of::Tree test]]|format=tree|parent=Has parent}}"
+		},
+		{
+			"page": "Example/Tree 01-02",
+			"contents": "{{#ask:[[Part of::Tree test]]|format=tree|parent=Has parent|root=Example/Tree 1}}"
+		}
+	],
+	"tests": [
+		{
+			"type": "parser",
+			"about": "Tree 01-01 (Multi-page result)",
+			"subject": "Example/Tree 01-01",
+			"assert-output": {
+				"to-contain": [
+					"<div class=\"srf-tree\">\n<ul><li><a href=\"/.*/Example/Tree_1\" title=\"Example/Tree 1\">Example/Tree 1</a>\n<ul><li><a href=\"/.*/Example/Tree_11\" title=\"Example/Tree 11\">Example/Tree 11</a></li>\n<li><a href=\"/.*/Example/Tree_12\" title=\"Example/Tree 12\">Example/Tree 12</a>\n<ul><li><a href=\"/.*/Example/Tree_121\" title=\"Example/Tree 121\">Example/Tree 121</a></li></ul></li>\n<li><a href=\"/.*/Example/Tree_13\" title=\"Example/Tree 13\">Example/Tree 13</a></li></ul></li>\n<li><a href=\"/.*/Example/Tree_2\" title=\"Example/Tree 2\">Example/Tree 2</a>\n<ul><li><a href=\"/.*/Example/Tree_21\" title=\"Example/Tree 21\">Example/Tree 21</a></li>\n<li><a href=\"/.*/Example/Tree_22\" title=\"Example/Tree 22\">Example/Tree 22</a></li></ul></li>\n<li><a href=\"/.*/Example/Tree_3\" title=\"Example/Tree 3\">Example/Tree 3</a>\n<ul><li><a href=\"/.*/Example/Tree_31\" title=\"Example/Tree 31\">Example/Tree 31</a></li>\n<li><a href=\"/.*/Example/Tree_32\" title=\"Example/Tree 32\">Example/Tree 32</a>\n<ul><li><a href=\"/.*/Example/Tree_321\" title=\"Example/Tree 321\">Example/Tree 321</a></li></ul></li>\n<li><a href=\"/.*/Example/Tree_33\" title=\"Example/Tree 33\">Example/Tree 33</a></li></ul></li></ul>\n</div>"
```

Comment:
In fact, I was about to implement "do-match" and "not-match" only to discover that it exists already. :)